includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -574,6 +722,19 @@ public function get_error_message( $error_code ) {
 				$message = __( 'Unable to receive access token because of an unsupported grant type.', 'google-site-kit' );
 				break;
 			default:
+				if ( $this->using_proxy() ) {
+					$access_code = $this->user_options->get( self::OPTION_PROXY_ACCESS_CODE );
+					if ( ! empty( $access_code ) ) {
+						$message = sprintf(
+							/* translators: 1: error code from API, 2: URL to re-authenticate */
+							__( 'Setup Error (code: %1$s). <a href="%2$s">Re-authenticate with Google</a>', 'google-site-kit' ),
+							$error_code,
+							$this->get_proxy_setup_url( $access_code )
```

Comment:
Should use `esc_url`

```suggestion
							esc_url( $this->get_proxy_setup_url( $access_code ) )
```